### ISSUE TITLE
Feature suggestion: Allow social cards to use custom fonts

### ISSUE THREAD

**Issue by hellt** [squidfunk/mkdocs-material#3247]
### Contribution guidelines

- [X] I've read the [contribution guidelines](https://github.com/squidfunk/mkdocs-material/blob/master/CONTRIBUTING.md) and wholeheartedly agree

### I want to suggest an idea and checked that ...

- [X] ... to my best knowledge, my idea wouldn't break something for other users
- [X] ... the documentation does not mention anything about my idea
- [X] ... there are no open or closed issues that are related to my idea

### Description

Social cards can currently work for sites that use only Google font name set in the `.theme.font.text` config parameter.

If the documentation portal happens to use a custom font sourced not from Google Fonts but, say, included via custom CSS, then social cards won't work.

This proposal suggests adding a `cards_font` config setting for the `socials` config region:

```yaml
theme:
  font:
    text: myCustomFont

plugins:
  - social:
      cards_font: assets/fonts/myCustomFont.ttf # if a font ends with .ttf - this is a local font
      # cards_font: Roboto # or refer to a Google-Font available font
```

The font can either be provided with a path to a `ttf` file or represent a name of a font that is available in Google Fonts.

### Use Cases

This feature makes it possible to generate social cards for doc sites that use custom, non-Google Fonts font.

**Comment by squidfunk** [comment]
As discussed in #3246, `cards_font` should be a path to a user-provided `*.ttf`, as this allows for more flexibility. This would also mean that custom Google Fonts are supported, as they can be downloaded as `*.ttf` files.

**Comment by squidfunk** [comment]
Fixed in [`d852efc99`](https://github.com/squidfunk/mkdocs-material-insiders/commit/d852efc996e86ce4ad1cd4ff5d9eb8b1554cb7ad). The `cards_font` setting can now be set to any valid Google Font:

``` yaml
plugins:
  - social:
      cards_font: Titillium Web
```

After investigating possible solutions, I've decided against supporting custom `*.ttf` files, because it makes the plugin much more complicated and I don't think that many users will benefit from this. Google Fonts really offers a lot of fonts, so if the exact font is not available, the recommendation is to choose one that is similar.

**Comment by hellt** [comment]
thanks @squidfunk

**Comment by squidfunk** [comment]
Released as part of 8.0.4+insiders-4.3.0!
